services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py

```diff
@@ -341,11 +341,18 @@ async def upsert_tasks_from_project(
                     if str(comp_task_db.node_id) not in published_nodes
                     else set()
                 )
+                update_values = (
+                    {"progress": None}
+                    if f"{comp_task_db.node_id}" in published_nodes
```

Comment:
are you sure? of the type here?
NOTE that `published_nodes: list[NodeID]` . And typically `NodeID` is not a string!

Reply:
STYLE:  IMO it reads better as foolws

```python
update_values = {} # i.e. nothing to update

if comp_task_db.node_id in published_nodes:
     update_values["progress"] = None
     
if to_node_class(comp_task_db.image.name) != NodeClass.FRONTEND:
     exclusion_rule.add("outputs")
     update_values.clear() # forces no udpate

on_update_stmt = ... | update_values

```

Reply:
good point with the type will check